include/logging/log_core.h

```diff
@@ -402,9 +402,12 @@ static inline char z_log_minimal_level_to_char(int level)
 #define LOG_FILTER_FIRST_BACKEND_SLOT_IDX 1
 
 #ifdef CONFIG_LOG_RUNTIME_FILTERING
+#define LOG_CHECK_CTX_LVL_FILTER(ctx, _level, _filter) \
```

Comment:
why not move it one line above and keep common for all cases ? `LOG_RUNTIME_FILTER` has 2 versions depending on runtime filtering being on of off so it should be ok.

Reply:
Problem was that when runtime filtering is not used, `(_level <= LOG_RUNTIME_FILTER(_filter))` is reduced to `(_level <= LOG_LEVEL_DBG)` -> `(_level <= 4)` -> `1`, so it has no effect.

Coverity then complains that `(is_user_context || 1)` disregards the first part.